korlibs-string/src@apple/korlibs/io/lang/CharsetApple.kt

```diff
@@ -0,0 +1,48 @@
+package korlibs.io.lang
+
+import korlibs.memory.ByteArrayBuilder
+import kotlinx.cinterop.ExperimentalForeignApi
+import kotlinx.cinterop.UnsafeNumber
+import platform.CoreFoundation.CFStringConvertEncodingToNSStringEncoding
+import platform.CoreFoundation.CFStringConvertIANACharSetNameToEncoding
+import platform.CoreFoundation.kCFStringEncodingInvalidId
+import platform.Foundation.NSString
+import platform.Foundation.NSStringEncoding
+import platform.Foundation.create
+import platform.Foundation.dataUsingEncoding
+
+@OptIn(UnsafeNumber::class, ExperimentalForeignApi::class)
+class CharsetApple(name: String) : Charset(name) {
+    private val encoding: NSStringEncoding = getEncoding(name)
+
+    private fun getEncoding(name: String): NSStringEncoding {
+        val encoding = name.uppercase().useCFStringRef { CFStringConvertEncodingToNSStringEncoding(CFStringConvertIANACharSetNameToEncoding(it)) }
+        if (encoding.toUInt() != kCFStringEncodingInvalidId) {
+            return encoding
+        } else {
+            throw IllegalArgumentException("Charset $name is not supported by apple.")
+        }
```

Comment:
Not sure, maybe `autoreleasepool { }` would be required.

Reply:
@soywiz I'm also not sure about that, but I think `autoreleasepool { }` is only required if using intensive object creation code or creating too many objects.